hafnian/__init__.py

```diff
@@ -81,11 +81,13 @@
     haf_rpt_real,
     haf_rpt_complex,
     reduction,
+    perm_complex,
+    perm_real,
```

Comment:
Similarly to comment below, if `perm_complex` and `perm_real` are not being used in `_hafnian.py`, they can be deleted there (and here)